crates/wasmtime/src/runtime/type_registry.rs

```diff
@@ -812,6 +853,11 @@ impl TypeRegistryInner {
                     }
                 }
 
+                // Remove the type's supertypes list, if any.
+                if self.type_to_supertypes.get(ty).is_some_and(|s| s.is_some()) {
+                    self.type_to_supertypes[ty] = None;
+                }
```

Comment:
Could this always do `= None` regardless of whether there's an array there or not?

Reply:
Unfortunately always doing `self.types_to_supertypes = None` would force the secondary map to allocate even when we aren't ever using Wasm GC. I could remove the `is_some_and(..)` suffix tho; I'll do that.

Reply:
Ah ok makes sense, mind leaving a comment that the purpose here is to avoid touching the map entirely?